modules/apps/account/account-service/src/main/java/com/liferay/account/service/impl/AccountRoleLocalServiceImpl.java

```diff
@@ -230,18 +230,23 @@ public AccountRole getOrAddEmptyAccountRole(
 		return _emptyModelManager.getOrAddEmptyModel(
 			AccountRole.class, companyId,
 			() -> {
-				Role role = _roleLocalService.getOrAddEmptyRole(
-					externalReferenceCode, companyId, userId,
-					AccountRole.class.getName(),
-					AccountConstants.ACCOUNT_ENTRY_ID_DEFAULT, name,
-					RoleConstants.TYPE_ACCOUNT);
+				try {
+					Role role = _roleLocalService.getOrAddEmptyRole(
```

Comment:
why cant we modify the signature of the method from `Exception` to `PortalException`?
(instead of adding the try/catch)